lib/server.js

```diff
@@ -92,6 +92,19 @@ class Server {
     } = prepareApplications(applications));
   }
 
+  // Update applivation version for each connection.
+  updateConnections() {
+    for (const connection of this.clients.values()) {
+      if (semver.valid(connection.requestedVersion)) {
```

Comment:
It's better to add a comment that this function will not update clients who specified direct versions and not version-range upon request in order to make this clearer. Or just comment here like 'ignore clients who requested specific API versions'.